crates/sui-json-rpc/src/axum_router.rs

```diff
@@ -32,15 +38,29 @@ pub struct JsonRpcService<L> {
     /// Registered server methods.
     methods: Methods,
     rpc_router: RpcRouter,
+    traffic_controller: Option<Arc<TrafficController>>,
 }
 
 impl<L> JsonRpcService<L> {
-    pub fn new(methods: Methods, rpc_router: RpcRouter, logger: L) -> Self {
+    pub async fn new(
+        methods: Methods,
+        rpc_router: RpcRouter,
+        logger: L,
+        remote_fw_config: RemoteFirewallConfig,
+        policy_config: Option<PolicyConfig>,
+    ) -> Self {
         Self {
             methods,
             rpc_router,
             logger,
             id_provider: Arc::new(RandomIntegerIdProvider),
+            traffic_controller: if let Some(policy) = policy_config {
+                Some(Arc::new(
+                    TrafficController::spawn(remote_fw_config, policy).await,
```

Comment:
I'm not sure i understand why spawning something would require this to be async. any way we could make this function sync and do any of the async stuff in the presumably spawned task in the background?